### PR TITLE
Puppetclasses distinct order by - removed ()

### PR DESCRIPTION
**Pull request by isratrade**

### CODE REVIEW

File: test/fixtures/roles.yml [hunk 1]
```diff
@@ -180,7 +180,7 @@ create_hosts:
     - :create_hosts
 
 none_compute_resources:
-  name: No right on compute resources and vms
+  name: No rights
```
Comment (skottler) [inline test/fixtures/roles.yml:183]: Was this intended to be changed?

Comment (isratrade) [inline test/fixtures/roles.yml:183]: Yes, it caused a failure when using postgres as the test database.  
On the role.rb model, there is a length contraint:   validates_length_of :name, :maximum => 30
The name in the fixture was 37 characters, so I just shortened it.
I assume other databases just truncated the name with no problem.

Comment (skottler) [inline test/fixtures/roles.yml:183]: I see. Cool, then looks really to get merged!

File: app/models/puppetclass.rb [hunk 2]
```diff
@@ -20,7 +20,7 @@ class Puppetclass < ActiveRecord::Base
 
   before_destroy EnsureNotUsedBy.new(:hosts)
   before_destroy EnsureNotUsedBy.new(:hostgroups)
-  default_scope :order => 'LOWER(puppetclasses.name)'
+  default_scope :order => 'puppetclasses.name'
```
Comment (ohadlevy) [inline app/models/puppetclass.rb:23]: i think we need the same fix for lookup_keys?

**Comment by ohadlevy** [comment]
i rebased as this pull request contained multiple commits that are already in developed, and amended the patch to include environments and lookup_\* as well. thanks!

**Pull request closed without merging**